process/block/baseProcess.go

```diff
@@ -652,6 +657,8 @@ func (bp *baseProcessor) cleanupPools(headerHandler data.HeaderHandler) {
 	} else {
 		bp.cleanupPoolsForCrossShard(core.MetachainShardId, noncesToFinal)
 	}
+
+	bp.transactionsLogProcessor.Clean()
```

Comment:
This should have gone to transactionCoordinator component.
There you have CreateBlockStarted and everything.

Reply:
I agree. I always encouraged cleaning up first then started processing stuff instead of processing stuff and then cleaning up as there might be situations (or future situations) when, due to an early exit, the cleanup never happens.

Reply:
done. good suggestion. :+1: